File: EulerProject/0xx/00x/002/main.cs
using System;
using System.Collections.Generic;

static class Fib {
    static public void fib(ref long cur, ref long next) {
        next += cur;
        cur = next - cur;
    }
    static public void Main() {
        long a = 1, b = 1;
        long sum = 0;
        while (b < 4e6) {
            fib(ref a, ref b);
            if (b % 2 == 0)
                sum += b;
        }
        Console.WriteLine(sum);
    }
}

File: EulerProject/0xx/00x/002/main.c
#include <stdio.h>

#define fib(cur, next)                                                         \
    {                                                                          \
        next += cur;                                                           \
        cur = next - cur;                                                      \
    }

int main() {
    int a = 1, b = 1;
    int sum = 0;
    while (b < 4e6) {
        fib (a, b);
        if (b % 2 == 0)
            sum += b;
    }
    printf ("%d\n", sum);
}
